core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -609,7 +616,7 @@ object Cause extends Serializable {
       new Both(left, right)
   }
 
-  private final case class Data(stackless: Boolean)
```

Comment:
Does it ever make sense to have a stackful, silent cause? If not, I propose to use a custom coproduct here for better readability

Reply:
Do you mean something like that?
```scala
sealed trait Data
case class Stackless(value: Boolean) extends Data
case class Silent(value: Boolean) extends Data
```

Reply:
Or even
```
sealed trait Data
case object Stackless extends Data
case object Silent extends Data
```

Reply:
Yeah, like this 👍

Reply:
One thing we're losing here is the ability to "turn on" and "turn off" stack traces. In the previous implementation stackless was compositional with inner scopes taking precedence (like almost everything else), so high up in the Cause tree  someone could say I don't want to show stack traces but then lower down someone could say no I really do want to show the trace for this cause.

Right now we don't need to do that for `silent` because we are only calling it on simple causes but there may be value in that in the future. There is also some value in having all `Meta` options work the same way. So I would suggest we have `final case class Data(stackless: Boolean, silent: Boolean)`.

Reply:
Loss of compositional power is a serious red flag...

Reply:
@jdegoes If we do what I am suggesting we will maintain the current compositional power of `stackless` and `silent` will compose the same way.

Reply:
@adamgfraser Good point. In that case I agree that the previous encoding was superior